woocommerce/compatibility/class-sv-wc-order-compatibility.php

```diff
@@ -522,6 +524,200 @@ public static function get_edit_order_url( \WC_Order $order ) {
 	}
 
 
+	/**
+	 * Gets the admin screen ID for orders.
+	 *
+	 * This method detects the expected orders screen ID according to HPOS availability.
+	 * `shop_order` as a registered post type as the screen ID is no longer used when HPOS is active.
+	 *
+	 * @since x.y.z
+	 *
+	 * @return string
+	 */
+	public static function get_screen_id( ) : string {
+
+		return SV_WC_Plugin_Compatibility::is_hpos_enabled()
+			? wc_get_page_screen_id( 'shop-order' )
+			: 'shop_order';
+	}
+
+
+	/**
+	 * Determines whether a given identifier is a WooCommerce order or not, according to HPOS availability.
+	 *
+	 * @since x.y.z
+	 *
+	 * @param int|\WP_Post|\WC_Order|null $post_order_or_id identifier of a possible order
+	 * @param string|string[] $order_type the order type, defaults to shop_order, can specify multiple types
+	 * @return bool
+	 */
+	public static function is_order( $post_order_or_id, $order_type = 'shop_order' ) : bool {
+
+		if ( ! SV_WC_Plugin_Compatibility::is_hpos_enabled() ) {
+
+			if ( $post_order_or_id instanceof \WC_Order ) {
+				$post_order_or_id = $post_order_or_id->get_id();
+			}
+
+			$found_type = is_numeric( $post_order_or_id ) || $post_order_or_id instanceof \WP_Post ? get_post_type( $post_order_or_id ) : null;
+
+		} else {
+
+			$found_type = OrderUtil::get_order_type( $post_order_or_id );
+		}
+
+		return $found_type && ( is_string( $order_type ) && $found_type === $order_type ) || ( is_array( $order_type ) && in_array( $found_type, $order_type, true ) );
+	}
+
+
+	/**
+	 * Determines whether a given identifier is a WooCommerce refund or not, according to HPOS availability.
+	 *
+	 * @since x.y.z
+	 *
+	 * @param int|\WP_Post|\WC_Order|null $order_post_or_id identifier of a possible order
+	 * @return bool
+	 */
+	public static function is_refund( $order_post_or_id ) : bool {
+
+		return static::is_order( $order_post_or_id, 'shop_order_refund' );
+	}
+
+
+	/**
+	 * Gets the order meta according to HPOS availability.
+	 *
+	 * Uses {@see \WC_Order::get_meta()} if HPOS is enabled, otherwise it uses the WordPress {@see get_post_meta()} function.
+	 *
+	 * @since x.y.z
+	 *
+	 * @param int|\WC_Order $order order ID or object
+	 * @param string $meta_key meta key
+	 * @param bool $single return the first found meta with key (true), or all meta sharing the same key (default true)
+	 * @return mixed
+	 */
+	public static function get_order_meta( $order, string $meta_key, bool $single = true ) {
+
+		if ( SV_WC_Plugin_Compatibility::is_hpos_enabled() ) {
+
+			$value = $single ? '' : [];
+			$order = ! $order instanceof \WC_Order && is_numeric( $order ) ? wc_get_order( (int) $order ) : $order;
```

Comment:
@sahmed2-godaddy just wondering if there's a reason why this conditional includes both `! $order instanceof \WC_Order` and `is_numeric( $order )` ? If `is_numeric( $order )` is true, then by definition `! $order instanceof \WC_Order` is also true.

Why not just check `is_numeric( $order )` and return `wc_get_order( (int) $order )` if true, otherwise just returning `$order`?

Reply:
@jstern-godaddy agreed, @unfulvio-godaddy made this change in a recent commit so unless he has a specific reason behind this change, I will update it.

Reply:
agreed it's redudant